### PR TITLE
Remove broadcast

### PR DESCRIPTION
**Pull request by lucasvanhalst**
```python
import json
import logging

from google.appengine.api import users
from google.appengine.ext import ndb
from solutions.common.migrations._009_nuke_broadcast import *


def run1():
    return nuke_broadcast_1()


def run2():
    return nuke_broadcast_2()


def run3():
    return nuke_broadcast_3()


def run4():
    return nuke_broadcast_4()


def run5():
    return nuke_broadcast_5()


def run6():
    return nuke_broadcast_6()


def run7():
    return nuke_broadcast_7()


def run8():
    return nuke_broadcast_8()


def run9():
    return nuke_broadcast_9()


def run_91():
    return nuke_broadcast_91()

```

### CODE REVIEW

**Comment by RubenMattan** [comment]
@lucasvanhalst add all migration in the description please :-)

File: src-test/rogerthat_tests/mobicage/service/test_system_api.py [hunk 1]
```diff
@@ -262,9 +262,7 @@ def test_menu_items(self):
         stream.seek(0)
         branding = store_branding(u'description', base64.b64encode(stream.read()))
 
-        put_broadcast_types(['News', 'Coupons'])
-
-        put_menu_item(icon_name, label, tag, [0, 1, 0], is_broadcast_settings=True, broadcast_branding=branding.id)
+        put_menu_item(icon_name, label, tag, [0, 1, 0])
```
Comment (RubenMattan) [inline src-test/rogerthat_tests/mobicage/service/test_system_api.py:265]: remove?

File: src/index.yaml [hunk 2]
```diff
@@ -40,78 +40,20 @@ indexes:
   - name: user
   - name: friend
 
-- kind: FriendServiceIdentityConnectionArchive
```
Comment (RubenMattan) [inline src/index.yaml:43]: restore?

File: src/index.yaml [hunk 3]
```diff
@@ -40,78 +40,20 @@ indexes:
...
-  properties:
-  - name: deleted
-  - name: service_identity_email
-
 - kind: FriendServiceIdentityConnection
   ancestor: yes
   properties:
-  - name: deleted
```
Comment (RubenMattan) [inline src/index.yaml:51]: what changed?

File: src/rogerthat/bizz/forms/__init__.py [hunk 4]
```diff
@@ -118,12 +120,12 @@ def delete_form_submissions(service_user, form_id, validate=True):
...
 @arguments(service_user=users.User, form_id=(int, long), testers=[users.User])
 def test_form(service_user, form_id, testers):
-    # type: (users.User, int, list[users.User]) -> None
+    # type: (users.User, int, List[users.User]) -> None
     form = get_form(form_id, service_user)
     request = TestFormRequestTO(id=form_id, version=form.version)
     to_put = []
     prof = get_service_profile(service_user)
-    branding = prof.broadcastBranding
+    branding = get_service_identity().menuBranding
```
Comment (RubenMattan) [inline src/rogerthat/bizz/forms/__init__.py:128]: missing service_user

File: src/rogerthat/bizz/friends.py [hunk 5]
```diff
@@ -1023,21 +996,10 @@ def side_effects():
...
-
-            to_delete = [BroadcastSettingsFlowCache.create_key(to, from_)]
-
-            if fsic and fsic.disabled_broadcast_types:
-                # Logical delete the FSIC to keep the disabled broadcast types
-                fsic.deleted = True
-                fsic.put()
-            else:
-                to_delete.append(fsic_key)
-            db.delete(to_delete)
```
Comment (RubenMattan) [inline src/rogerthat/bizz/friends.py:1040]: delete was removed

File: src/rogerthat/bizz/job/update_friends.py [hunk 6]
```diff
@@ -117,118 +114,78 @@ def schedule_update_friends_of_profile_info(profile_info, changed_properties=Non
...
         service_profile_or_service_user = get_service_profile(service_user, False)
     else:
         service_profile_or_service_user = service_user
-    if not force and not _must_continue_with_update_service(service_profile_or_service_user, False,
-                                                            clear_broadcast_settings_cache):
+    if not force and not _must_continue_with_update_service(service_profile_or_service_user, False):
         return
-    deferred.defer(_run_update_friend_for_service_identity, service_identity_user,
-                   clear_broadcast_settings_cache,
+    deferred.defer(_serialize_and_update_friend_for_service_identity, service_identity_user,
```
Comment (RubenMattan) [inline src/rogerthat/bizz/job/update_friends.py:157]: function changed

File: src/rogerthat/bizz/service/__init__.py [hunk 7]
```diff
@@ -2785,11 +2731,7 @@ def trans(updated_json_dict):
         current_mobile = users.get_current_mobile()
 
         if not friend_map:
-            if len(updated_json_dict) == 1 and DISABLED_BROADCAST_TYPES_USER_DATA_KEY in updated_json_dict:
-                # User is disabling a news item
-                friend_map = FriendMap.create(friend_user)
-            else:
-                raise FriendNotFoundException()
+            raise FriendNotFoundException()
```
Comment (RubenMattan) [inline src/rogerthat/bizz/service/__init__.py:2734]: hmmm this can trigger some errors with old clients

File: src/rogerthat/models/__init__.py [hunk 8]
```diff
@@ -1116,16 +1116,6 @@ def defaultLanguage(self):
     def service_user(self):
         return self.user
 
-    def addFlag(self, flag):
-        azzert(flag in ServiceProfile.FLAGS)
-        self.flags = set_flag(flag, self.flags)
```
Comment (RubenMattan) [inline src/rogerthat/models/__init__.py:1121]: can we remove the flags property?

File: src/rogerthat/models/__init__.py [hunk 9]
```diff
@@ -2591,7 +2499,6 @@ class ServiceMenuDef(db.Model):
     link = db.TextProperty()
     isExternalLink = db.BooleanProperty(indexed=False, default=False)
     requestUserLink = db.BooleanProperty(indexed=False, default=False)
-    isBroadcastSettings = db.BooleanProperty(indexed=True)
```
Comment (RubenMattan) [inline src/rogerthat/models/__init__.py:2594]: will this model be removed during the migration or publish?

File: src/index.yaml (hunk 3 above)
Comment (lucasvanhalst) [reply src/index.yaml:51]: I removed all of them and rebuild them manually

get_friend_service_identity_connections_keys_query uses:
```
  ancestor: yes
  properties:
  - name: service_identity_email
```
the rest of the queries that require composite indexes can be found on the FriendServiceIdentityyConnection model itself as classmethods

File: src/index.yaml (hunk 2 above)
Comment (lucasvanhalst) [reply src/index.yaml:43]: Not needed since the deleted property was removed

File: src/rogerthat/bizz/job/update_friends.py (hunk 6 above)
Comment (lucasvanhalst) [reply src/rogerthat/bizz/job/update_friends.py:157]: that's intentional, _run_update_friend_for_service_identity has been removed as it called _serialize_and_update_friend_for_service_identity and did nothing different anymore after the changes

File: src/rogerthat/models/__init__.py (hunk 8 above)
Comment (lucasvanhalst) [reply src/rogerthat/models/__init__.py:1121]: looks like I forgot to do that

File: src/rogerthat/models/__init__.py (hunk 9 above)
Comment (lucasvanhalst) [reply src/rogerthat/models/__init__.py:2594]: migration

File: src/solutions/common/bizz/provisioning.py [hunk 10]
```diff
@@ -117,7 +116,7 @@
     SolutionModule.AGENDA: POKE_TAG_EVENTS,
     SolutionModule.APPOINTMENT: POKE_TAG_APPOINTMENT,
     SolutionModule.ASK_QUESTION: POKE_TAG_ASK_QUESTION,
-    SolutionModule.BROADCAST: ServiceMenuDef.TAG_MC_BROADCAST_SETTINGS,
+    SolutionModule.NEWS: POKE_TAG_BROADCAST_CREATE_NEWS,
```
Comment (RubenMattan) [inline src/solutions/common/bizz/provisioning.py:119]: hmm don't know if this is correct. Should this not be None?

File: src/solutions/common/bizz/provisioning.py [hunk 11]
```diff
@@ -689,12 +680,11 @@ def menu_sorting_key(ssmi):
         if coords[2] >= last_prefered_page and ssmi_dict["preferred_page"] != -1:
             last_prefered_page = coords[2] + 1
 
-        logging.debug("Putting menu item at %s: %s",
-                      coords, serialize_complex_value(ssmi, SolutionServiceMenuItem, False))
+        logging.debug("Putting menu item at %s: %s", coords, ssmi)
```
Comment (RubenMattan) [inline src/solutions/common/bizz/provisioning.py:683]: does this log the content of ssmi?

Comment (lucasvanhalst) [reply src/solutions/common/bizz/provisioning.py:683]: it inherits from TO so yeah

File: src/solutions/common/bizz/provisioning.py (hunk 10 above)
Comment (lucasvanhalst) [reply src/solutions/common/bizz/provisioning.py:119]: doesn't really matter

File: src/rogerthat/models/__init__.py (hunk 9 above)
Comment (RubenMattan) [reply src/rogerthat/models/__init__.py:2594]: doet the migration work if this property is removed or does this needs to happen on another version?

**Comment by RubenMattan** [comment]
LGTM

**Pull request merged**